proposals/0243-character-literals.md

```diff
@@ -0,0 +1,107 @@
+
+ ## Source compatibility
+
+As the use of the new single quoted syntax is opt-in existing code will continue to compile as before i.e. the proposed implementation is not source breaking. Only where the user has opted to use the new single quoted spelling will the integer conversions be available. This is possible to add a warning and fix-it to prompt the user to move to the new syntax in the course of time. In practice, the Character and Unicode.Scalar types occur do not occur frequently in code so this would not be an arduous migration.
+
+ ## Effect on ABI stability
+
+ Assuming injecting `@_marker` protocols does not alter witness table layout and ABI, this is a purely lexer- and type checker-level change which does not affect the storage or entry points of `Character` and `Unicode.Scalar`. These initializers are an implementation artifact of Swift's protocol-driven literals model, and are meant to be automatically invoked by the compiler. As they are not intended for “public consumption”, we see no reason to continue having the compiler invoke them.
+
+ ## Effect on API resilience
+
+ This is a purely lexer- and type checker-level change which does not affect the API of the standard library.
+
+ ## Alternatives considered
+
+ The most obvious alternative is to simply leave things the way they are where double quoted `String` literals can perform service as `Character` or `UnicodeScalar` values as required. At its heart, while this is transparent to users, this devalues the role of `Characters` in source code — a distinction that may come in handy working in lower-level code. 
+
+ Another alternative discussed on [another thread](https://forums.swift.org/t/unicode-scalar-literals/22224) was “Unicode Scalar Literals”. Unicode scalar literals would have the benefit of allowing concise access to codepoint and ASCII APIs, as methods and properties could be accessed from `'a'` expressions instead of unwieldy `('a' as Unicode.Scalar)` expressions. However the authors feel this would contradict Swift’s `String` philosophy, which explicitly recognizes `Character` as the natural element of `String`, not `Unicode.Scalar`. In the end this is configurable in the implementataion.
```

Comment:
i don’t think “configurable in the implementation” is relevant here. 

however: i would suggest adding a *Future Directions* section discussing things like multi-character single-quoted literals, like `'abc'` that could be used to initialize collections of `UInt8`.